### PR TITLE
build omnisharp-roslyn from source

### PR DESCRIPTION
**Pull request by corngood**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 126487` at c5b11e2e run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>omnisharp-roslyn</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/development/tools/omnisharp-roslyn/default.nix:88:3:
```
   |
88 |   installPhase = ''
   |   ^
```
</li>
  </ul>
</details>

**Comment by hpfr** [comment]
Is this still a WIP or can it be reviewed? This got omnisharp working for me on NixOS.

**Comment by corngood** [comment]
I've been using it for a while, and I'm happy with it.  Removed wip.

File: pkgs/development/tools/omnisharp-roslyn/default.nix [hunk 1]
```diff
@@ -1,42 +1,104 @@
...
+      <fallbackPackageFolders>
+        ${lib.concatStringsSep "\n" (map (package: "<add key=\"${package}\" value=\"${package}/lib/dotnet\"/>") deps)}
+      </fallbackPackageFolders>
+    </configuration>
+  '';
+
+in stdenv.mkDerivation rec {
 
   pname = "omnisharp-roslyn";
   version = "1.37.8";
```
Comment (ericdallo) [inline pkgs/development/tools/omnisharp-roslyn/default.nix:69]: There is 1.37.12 already, maybe bump after this PR

**Comment by ericdallo** [comment]
Result of `nixpkgs-review pr 126487` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>omnisharp-roslyn</li>
  </ul>
</details>

File: pkgs/development/tools/omnisharp-roslyn/default.nix [hunk 2]
```diff
@@ -1,42 +1,104 @@
...
+    ln -s ${msbuild}/lib/mono/msbuild/Current/bin $out/src/.msbuild/Current/Bin
 
-    makeWrapper ${mono}/bin/mono $out/bin/omnisharp \
-    --add-flags "$out/src/OmniSharp.exe"
+    makeWrapper ${mono6}/bin/mono $out/bin/omnisharp \
+      --prefix PATH : ${dotnet-sdk}/bin \
+      --add-flags "$out/src/OmniSharp.exe"
   '';
 
   meta = with lib; {
```
Comment (ericdallo) [inline pkgs/development/tools/omnisharp-roslyn/default.nix:104]: Willing to add yourself as a maintainer? :)

**Review by ericdallo — APPROVED** [review]
Looks good and works for me :)
Thanks for this!

File: pkgs/development/tools/omnisharp-roslyn/default.nix (hunk 1 above)
Comment (corngood) [reply pkgs/development/tools/omnisharp-roslyn/default.nix:69]: Will do

File: pkgs/development/tools/omnisharp-roslyn/default.nix (hunk 2 above)
Comment (corngood) [reply pkgs/development/tools/omnisharp-roslyn/default.nix:104]: Sure, done.

**Comment by ericdallo** [comment]
Also, pease squash commits into a single one

**Comment by corngood** [comment]
The commits all solve different problems, and all build correctly, so I'm not sure they should be squashed.  However, I've pushed a squashed version, if you prefer it that way.

**Comment by ericdallo** [comment]
It's not a personal preference, but recommended by nixpkgs contributing docs. Thanks!
@SuperSandro2000 could you merge this please?

**Pull request merged**